MapleServer2/Data/Static/SkillMetadataStorage.cs

```diff
@@ -1,28 +1,63 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.IO;
 using Maple2Storage.Types.Metadata;
+using MapleServer2.Constants.Skills;
+using MapleServer2.Enums;
 using ProtoBuf;
 
 namespace MapleServer2.Data.Static
 {
     public static class SkillMetadataStorage
     {
-        private static readonly Dictionary<int, List<SkillLevel>> skillLevel = new Dictionary<int, List<SkillLevel>>();
+        private static readonly Dictionary<int, SkillMetadata> skill = new Dictionary<int, SkillMetadata>();
 
         static SkillMetadataStorage()
         {
             using FileStream stream = File.OpenRead("Maple2Storage/Resources/ms2-skill-metadata");
             List<SkillMetadata> skillList = Serializer.Deserialize<List<SkillMetadata>>(stream);
             foreach (SkillMetadata skills in skillList)
             {
-                skillLevel.Add(skills.SkillId, skills.SkillLevel);
+                skill[skills.SkillId] = skills;
             }
         }
 
-        // TODO: Get Skills from JobId
-        public static IEnumerable<SkillLevel> GetSkills(int level)
+        public static SkillMetadata GetSkill(int id)
         {
-            return skillLevel.GetValueOrDefault(level);
+            return skill.GetValueOrDefault(id);
+        }
+
+        public static List<SkillMetadata> GetJobSkills(Job job = Job.None)
+        {
+            List<SkillMetadata> jobSkill = new List<SkillMetadata>();
+            int jobCode = ((Convert.ToInt32(job) / 100) + 100) * 100000; // 10X0000 10100000
```

Comment:
Probably should specify the skill ids are in this format because otherwise it's not obvious at all

Reply:
Didn't get what you mean on this, could you elaborate more ?